File: release/ProjectGamma/desktop-app/ai_suggest_task.test.js
/**
 * TDD · heuristische KI-Suggestion fuer "Idee -> Aufgabe" auf dem Desktop.
 * Spiegelt mobile-app/.../ai_suggest_task.dart (snake_case-modul, pure JS).
 */
'use strict';
const test = require('node:test');
const assert = require('node:assert/strict');
const { aiSuggestTask } = require('./ai_suggest_task.js');

test('leere oder whitespace-eingabe -> null', () => {
  assert.equal(aiSuggestTask(''), null);
  assert.equal(aiSuggestTask('   '), null);
});

test('einfacher text -> title getrimmt + erstes zeichen gross', () => {
  const s = aiSuggestTask('einkaufen morgen');
  assert.equal(s.title, 'Einkaufen morgen');
  assert.equal(s.meta, 'aus idee');
  assert.equal(s.priority, 3);
});

test('hoch-keyword -> priority 4 + meta "hoch"', () => {
  const s = aiSuggestTask('DRINGEND backup machen');
  assert.equal(s.priority, 4);
  assert.match(s.meta, /hoch/i);
  assert.ok(!/^dringend/i.test(s.title));
});

test('niedrig-keyword -> priority 2', () => {
  const s = aiSuggestTask('vielleicht mal die docs lesen');
  assert.equal(s.priority, 2);
  assert.match(s.meta, /niedrig/i);
});

test('langer text wird auf 80 zeichen gekuerzt', () => {
  const s = aiSuggestTask('a'.repeat(200));
  assert.ok(s.title.length <= 80);
});

File: mobile-app/test/features/idea_to_task/ai_suggest_task_test.dart
// TDD: heuristische KI-Vorschlaege fuer Idee->Aufgabe (1-Tap-Flow).
import 'package:flutter_test/flutter_test.dart';
import 'package:projectgamma_mobile/features/idea_to_task/ai_suggest_task.dart';

void main() {
  group('aiSuggestTask', () {
    test('leere oder whitespace-eingabe -> null', () {
      expect(aiSuggestTask(''), isNull);
      expect(aiSuggestTask('   '), isNull);
    });

    test('einfacher text -> title getrimmt + erstes zeichen gross', () {
      final s = aiSuggestTask('einkaufen morgen')!;
      expect(s.title, 'Einkaufen morgen');
      expect(s.meta, 'aus idee');
      expect(s.priority, 3);
    });

    test('hoch-keyword -> priority 4 + meta enthaelt "hoch"', () {
      final s = aiSuggestTask('DRINGEND backup machen')!;
      expect(s.priority, 4);
      expect(s.meta.toLowerCase(), contains('hoch'));
      expect(s.title.toLowerCase(), isNot(startsWith('dringend')));
    });

    test('niedrig-keyword -> priority 2', () {
      final s = aiSuggestTask('vielleicht mal die docs lesen')!;
      expect(s.priority, 2);
      expect(s.meta.toLowerCase(), contains('niedrig'));
    });

    test('langer text wird bei 80 zeichen gekuerzt', () {
      final long = 'a' * 200;
      final s = aiSuggestTask(long)!;
      expect(s.title.length, lessThanOrEqualTo(80));
    });

    test('toJson liefert title/meta/priority', () {
      final j = aiSuggestTask('foo bar')!.toJson();
      expect(j['title'], 'Foo bar');
      expect(j['meta'], 'aus idee');
      expect(j['priority'], 3);
    });
  });
}
